api/plugins/session-auth.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  DoneFuncWithErrOrRes,
+  FastifyPluginCallback,
+  FastifyReply,
+  FastifyRequest
+} from 'fastify';
+import fp from 'fastify-plugin';
+
+const sessionAuth: FastifyPluginCallback = (fastify, _opts, done) => {
+  fastify.decorate('authenticateSession', authenticateSession);
+
+  function authenticateSession(
+    req: FastifyRequest,
+    res: FastifyReply,
+    done: DoneFuncWithErrOrRes
+  ) {
```

Comment:
Fastify has a type for this `onRequestHookHandler`, so you can use that like this:

```ts
  const authenticateSession: onRequestHookHandler = (req, res, done) => {
```

It should be fine to use a const since we don't need access to `this`.